flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -127,6 +130,15 @@
      */
     private final Map<Integer, SubtaskGatewayImpl> subtaskGatewayMap;
 
+    /**
+     * A map that manages a completable future for each subtask. It helps to guarantee that when the
+     * coordinator starts doing a checkpoint, it will not receive events from its subtasks anymore,
+     * until the checkpoint is completed or aborted. This map is only read or modified in
+     * checkpoint-related processes, and note that concurrent execution attempt is currently
+     * guaranteed to be disabled when checkpoint is enabled.
+     */
+    private final Map<Integer, CompletableFuture<Acknowledge>> acknowledgeCloseGatewayFutureMap;
```

Comment:
According to the semantics of this map, do we need to update this map whenever subtask execution is updated (e.g. in `subtaskReset(...)`) to make sure that this map's value is always consistent with its semantics?